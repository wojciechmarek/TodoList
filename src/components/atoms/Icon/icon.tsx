import { IconProps } from './icon.interface';
import { IconContainer, IconStyle } from './icon.styled';

export const Icon = (props: IconProps) => {
  const { icon, ...params } = props;

  return (
    <IconContainer {...params}>
      <IconStyle>{icon}</IconStyle>
    </IconContainer>
  );
};
