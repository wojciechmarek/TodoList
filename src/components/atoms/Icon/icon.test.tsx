import { configure, render } from '@testing-library/react';
import { Icon } from './icon';
import { IconProps } from './icon.interface';
import TestRenderer from 'react-test-renderer';

describe('Icon', () => {
  let iconProps: IconProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    iconProps = {
      icon: 'icon',
    };
  });

  it('should render', () => {
    const { baseElement } = render(<Icon {...iconProps} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match the snapshot', () => {
    const tree = TestRenderer.create(<Icon {...iconProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
