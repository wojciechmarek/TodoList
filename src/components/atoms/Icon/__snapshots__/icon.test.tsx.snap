// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  height: calc(100% * 0.618);
  width: calc(100% * 0.618);
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    icon
  </div>
</div>
`;
